src/test/resources/application.properties

```diff
@@ -0,0 +1,8 @@
+spring.datasource.username=springuser
```

Comment:
Oh, duh! No wonder my tests weren't working! Good catch!

Reply:
Also, the OAuth2 configuration is needed because, if you run integration tests without those 3 configurations, the `client-id`, `client-secret`, and `scopes` then Spring Security will not autoconfigure a client registration repository for us, and apparently that can't be `null`